airbyte-integrations/bases/standard-source-test/src/main/java/io/airbyte/integrations/standardtest/source/TestSource.java

```diff
@@ -225,22 +243,100 @@ public void testRead() throws Exception {
     });
   }
 
-  /**
-   * Verify that the integration overwrites the first sync with the second sync.
-   */
   @Test
-  public void testSecondRead() throws Exception {
-    final List<AirbyteMessage> recordMessagesFirstRun = runRead(CatalogHelpers.toDefaultConfiguredCatalog(getCatalog()))
-        .stream().filter(m -> m.getType() == Type.RECORD)
-        .collect(Collectors.toList());
-    final List<AirbyteMessage> recordMessagesSecondRun = runRead(CatalogHelpers.toDefaultConfiguredCatalog(getCatalog()))
-        .stream().filter(m -> m.getType() == Type.RECORD)
-        .collect(Collectors.toList());
+  public void testIdenticalFullRefreshes() throws Exception {
+    ConfiguredAirbyteCatalog configuredCatalog = withFullRefreshSyncModes(getConfiguredCatalog());
+    final List<AirbyteRecordMessage> recordMessagesFirstRun = filterRecords(runRead(configuredCatalog));
+    final List<AirbyteRecordMessage> recordMessagesSecondRun = filterRecords(runRead(configuredCatalog));
     // the worker validates the messages, so we just validate the message, so we do not need to validate
     // again (as long as we use the worker, which we will not want to do long term).
     assertFalse(recordMessagesFirstRun.isEmpty());
     assertFalse(recordMessagesSecondRun.isEmpty());
-    assertSameMessages(recordMessagesSecondRun, recordMessagesSecondRun);
+
+    assertSameRecords(recordMessagesFirstRun, recordMessagesSecondRun);
```

Comment:
bugfix, it was testing equality of the same list

Reply:
🤦